sw/device/lib/dif/dif_keymgr.h

```diff
@@ -0,0 +1,656 @@
+ *
+ *   bool is_idle_and_ok = (status_codes == kDifKeymgrStatusCodeIdle);
+ *
+ * See also: `dif_keymgr_status_code_t`.
+ */
+typedef uint8_t dif_keymgr_status_codes_t;
+
+/**
+ * Gets the operational status of key manager.
+ *
+ * @param keymgr A key manager handle.
+ * @param[out] status_codes Out-param for key manager status codes.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_result_t dif_keymgr_get_status_codes(
+    const dif_keymgr_t *keymgr, dif_keymgr_status_codes_t *status_codes);
+
+/**
+ * Gets the current state of key manager.
+ *
+ * @param keymgr A key manager handle.
+ * @param[out] state Out-param for current key manager state.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_result_t dif_keymgr_get_state(const dif_keymgr_t *keymgr,
+                                         dif_keymgr_state_t *state);
+
+/**
+ * Generates an identity seed.
+ *
+ * This function requests key manager to generate an identity seed using its
+ * current secret value. Clients must first verify that the operation was
+ * successful using `dif_keymgr_get_status_codes()` before reading the generated
+ * identity seed using `dif_keymgr_read_output()`.
+ *
+ * The generated seed can be used to generate an identity using an asymmetric
+ * KDF.
+ *
+ * @param keymgr A key manager handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_lockable_result_t dif_keymgr_generate_identity_seed(
+    const dif_keymgr_t *keymgr);
+
+/**
+ * Destination of a versioned key generation operation.
+ *
+ * Key manager can make the output of a versioned key generation operation
+ * available to software or sideload it directly to a peripheral device. When
+ * the destination is a peripheral device, the output of the operation is not
+ * visible to software and a different derivation constant is used for each
+ * peripheral.
+ */
+typedef enum dif_keymgr_versioned_key_dest {
+  /**
+   * Store the generated versioned key in software visible registers.
+   *
+   * The generated versioned key can be read by calling
+   * `dif_keymgr_read_output()` after verifying that the operation was
+   * successful using `dif_keymgr_get_status_codes()`.
+   */
+  kDifKeymgrVersionedKeyDestSw,
+  /**
+   * Sideload the generated versioned key to AES device.
+   */
+  kDifKeymgrVersionedKeyDestAes,
+  /**
+   * Sideload the generated versioned key to HMAC device.
+   */
+  kDifKeymgrVersionedKeyDestHmac,
+  /**
+   * Sideload the generated versioned key to KMAC device.
+   */
+  kDifKeymgrVersionedKeyDestKmac,
+} dif_keymgr_versioned_key_dest_t;
+
+/**
+ * Parameters for generating a versioned key.
+ */
+typedef struct dif_keymgr_versioned_key_params {
+  /**
+   * Destination of the generated versioned key.
+   *
+   * See also: `dif_keymgr_versioned_key_dest_t`.
+   */
+  dif_keymgr_versioned_key_dest_t dest;
+  /**
+   * Salt value to use for key generation.
+   */
+  uint8_t salt[16];
```

Comment:
@tjaychen, @alphan 

Since we are combining ID and salt (I assume), is it possible to make this 256B?

Reply:
Done.